### PR TITLE
creation of service object

### PR DESCRIPTION
**Pull request by zackforbing**
on Brian's recommendation, I would like to implement a LookingForService file that contains all the api calls. Currently, there are four API calls in the frontend app:
1. GetRecentJobs, on the Jobs Index
2. GetFilteredJobs, on the Jobs Index
3. GetSingleJob, on the Jobs Show
4. GetJobTechCount, on the Trends Index

this falls under refactoring. If anyone thinks this course of action isn't worth taking, let me know, and I will reevaluate.

### CODE REVIEW

**Comment by mattr-** [comment]
IMHO, service objects don't need to be components as components are generally for view pieces only.

**Comment by MsJennyGiraffe** [comment]
I like the idea of breaking things out so it's easier to pick up and easier to see where things are.
I'd like to pair with you on this if I can.

**Comment by zackforbing** [comment]
@mattr- good call. looks like I'll attempt to make this a plain javascript object.

File: lib/components/LookingForService.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import axios from 'axios';
+
+class LookingForService extends React.Component {
+  constructor() {
+    super();
```
Comment (pwentz) [inline lib/components/LookingForService.js]: @zackforbing I think this is a good time to make a `services` dir and abstract this API call into a plain `.js` file. Axios will return you a promise object so you can save the promise as a variable in your component, then run the `.then()` function to extract the data and either render the data as it is, or save it to `state`.

**Comment by zackforbing** [comment]
@MsJennyGiraffe  and I tag teamed this. this PR extracts the Axios API calls into a POJO that handles all of them in one place. In the future, if changes need to be made to the calls, none will be missed. 

I'm glad I did this, as it brought me to the `.call()` function, which I had not really used before now and was unclear on its inner workings. In this use, it allows us to pass `this` from the React component into the LookingForService object, so the calls can basically be made as they were originally written when they were included in the React components.

**Comment by matthewrpacker** [comment]
@zackforbing and @MsJennyGiraffe 
Great refactor.  I really like the use of `call()` in JobIndex / TrendsIndex and `bind()` in JobShow.  Also, I like how you exported your functions from LookingForService given that all functions are considered public and used in other files.  Nice work!

**Comment by Laner12** [comment]
@zackforbing @MsJennyGiraffe 
I think this is a very clean refactor. This PR adds a level of clarity to the structure of the app by consolidating all of the API functionality into one location. @Carmer is there a way to cut down on the repetition of the promises, or do they have to be directly after each API call?

**Comment by Automatic365** [comment]
This is super streamlined. I think something like this will be able to add smoother implementation of the search functions and filtering. 

@Laner12 
I could be wrong, but maybe it is possible to break out the repetitive parts into a function that takes in data as the argument?

**Comment by kbs5280** [comment]
As we've discussed, this is an improvement. Nice use of call and bind. I noticed we call one function in the service with `.call` and one with `.bind` and I'm wondering why that is?

This makes the code easier to read, but also more maintainable. 

I am going to merge this PR.

**Comment by zackforbing** [comment]
that's actually a typo, and it doesn't work. it should be `call`. I'll have a new commit up in a sec.

**Comment by zackforbing** [comment]
@Laner12 @Automatic365 also I talked to @Carmer about refactoring the promises, and the stuff he dug up about it actually made it LESS intuitive and harder to read, and since the `setState` is slightly different in each one, I think we are good to leave the promises where they are.

**Comment by Carmer** [comment]
I think the promises are fine. Not to many and I think trying to remove them would make things unessesarily complex

**Pull request merged**